crates/core/c8y_api/src/smartrest/operations.rs

```diff
@@ -50,23 +52,31 @@ impl Operations {
         None
     }
 
-    pub fn filter_by_topic(&self, topic_name: &str) -> Vec<(String, Operation)> {
+    pub fn filter_by_topics(
+        &self,
+        topics: Vec<String>,
+        prefix: &TopicPrefix,
+    ) -> Vec<(String, Operation)> {
         let mut vec: Vec<(String, Operation)> = Vec::new();
         for op in self.operations.iter() {
             match (op.topic(), op.on_fragment()) {
                 (None, Some(on_fragment)) => vec.push((on_fragment, op.clone())),
-                (Some(topic), Some(on_fragment)) if topic == topic_name => {
-                    vec.push((on_fragment, op.clone()))
+                (Some(ref mut topic), Some(on_fragment)) => {
+                    Self::replace_bridge_prefix_template(topic, prefix);
+                    if topics.contains(topic) {
+                        vec.push((on_fragment, op.clone()))
+                    }
                 }
                 _ => {}
             }
         }
         vec
     }
-
+    /// Get topics of operations that are delivered via JSON over MQTT
     pub fn topics_for_operations(&self) -> HashSet<String> {
         self.operations
             .iter()
+            .filter(|operation| operation.on_fragment().is_some())
```

Comment:
Oh don't limit it to the ones with `on_fragment`.. This function is used to extract topics also for SmartREST custom operation handlers.